src/renderer/services/ethereum/balances.ts

```diff
@@ -4,34 +4,49 @@ import { AssetETH, baseAmount } from '@xchainjs/xchain-util'
 import * as FP from 'fp-ts/lib/function'
 import * as O from 'fp-ts/lib/Option'
 import * as Rx from 'rxjs'
-import { Observable } from 'rxjs'
 import { mergeMap, shareReplay, catchError, startWith, debounceTime } from 'rxjs/operators'
 
 import { ETH_DECIMAL } from '../../helpers/assetHelper'
 import { triggerStream } from '../../helpers/stateHelper'
-import { BalanceRD } from '../clients'
+import { WalletBalanceLD } from '../clients'
 import { ApiError, ErrorId } from '../wallet/types'
 import { client$ } from './common'
 
 /**
  * Observable to load balances
  * If client is not available, it returns an `initial` state
  */
-const loadBalances$ = (client: Client): Observable<BalanceRD> =>
-  Rx.from(client.getBalance()).pipe(
-    mergeMap((balance) =>
-      Rx.of(
-        RD.success({
-          asset: AssetETH,
-          amount: baseAmount(balance.toString(), ETH_DECIMAL),
-          walletAddress: client.getAddress()
-        })
-      )
-    ),
-    catchError((error: Error) =>
-      Rx.of(RD.failure({ errorId: ErrorId.GET_BALANCES, msg: error?.message ?? '' } as ApiError))
-    ),
-    startWith(RD.pending)
+const loadBalances$ = (client: Client): WalletBalanceLD =>
+  FP.pipe(
+    O.tryCatch(() => client.getAddress()),
+    O.fold(
+      () =>
+        Rx.of(
+          RD.failure<ApiError>({
+            errorId: ErrorId.GET_BALANCES,
+            // TODO (@Veado) Add i18n
+            msg: 'Cant get address from client'
+          })
+        ),
+      (walletAddress) =>
+        Rx.from(client.getBalance()).pipe(
+          mergeMap((balance) =>
+            Rx.of(
+              RD.success({
+                asset: AssetETH,
+                amount: baseAmount(balance.toString(), ETH_DECIMAL),
+                walletAddress
+              })
+            )
+          ),
```

Comment:
not sure but maybe this will be more readable ?
```ts
O.map((walletAddress) =>
    FP.pipe(
      Rx.of(client.getBalance()),
      RxOp.map((balance) =>
        RD.success({
          asset: AssetETH,
          amount: baseAmount(balance.toString(), ETH_DECIMAL),
          walletAddress
        })
      )
    )
  ),
  RxOp.switchMap(
    O.getOrElse(() =>
      Rx.of(
        RD.failure<ApiError>({
          errorId: ErrorId.GET_BALANCES,
          // TODO (@Veado) Add i18n
          msg: 'Cant get address from client'
        })
      )
    )
  )
```

Reply:
Yep, it looks better. For now we can ignore it here, because `xchain-ethereum` will be updated soon and with this update `loadBalances$` will be gone here.